src/components/SubmitButton.vue

```
@@ -53,251 +86,287 @@
-                this.showError &&
-                (this.state === 'failure' || this.state === 'warning') &&
-                (Boolean(this.err) || this.showEmpty)
-            );
+    props: {
+        submit: {
+            type: Function,
+            required: true,
         },
 
-        variants() {
-            return {
-                default: this.default,
-                success: this.success,
-                failure: this.failure,
-                warning: this.warning,
-            };
+        filter: {
+            type: Function,
+            default: null,
         },
-    },
 
-    props: {
         id: {
             type: String,
             default: null,
         },
+
         tabindex: {
-            default: '0',
+            type: [Number, String],
+            default: 0,
         },
 
-        popoverPlacement: {
-            type: String,
-            default: 'top',
-        },
         disabled: {
             type: Boolean,
             default: false,
         },
-        label: {
-            type: [String, Boolean],
-            default: 'Submit',
-        },
+
         size: {
             type: String,
             default: 'md',
         },
-        delay: {
-            type: Number,
-            default: 1000,
-        },
-        default: {
+
+        variant: {
             type: String,
             default: 'primary',
         },
-        success: {
-            type: String,
-            default: 'success',
+
+        duration: {
+            type: Number,
+            default: 1500,
         },
-        failure: {
+
+        confirm: {
             type: String,
-            default: 'danger',
+            default: '',
         },
-        warning: {
+
+        popoverPlacement: {
             type: String,
-            default: 'warning',
+            default: 'top',
         },
-        showEmpty: {
+
+        waitAtLeast: {
+            type: Number,
+            default: 250,
+        },
+
+        showError: {
```

I think `showError` and `showWarning` should be removed. Only `showError` is used, and only once. It is used in FileUploader.vue, as we bubble the error event up (we do an `$emit('error', $event)` on the `error` event). We can do some filtering there, and then the props are not used, and I think hiding errors or warnings is an anti pattern anyway (just like the term user).

---

Yeah I agree. I'll look into it.